src/components/keyboard-nav.js

```diff
@@ -0,0 +1,93 @@
+import * as _ from 'lodash/fp'
+import { useRef } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import { toIndexPairs } from 'src/libs/utils'
+
+
+export const withArrowKeyNavigation = ({ onKeyDown, children }) => {
+  const count = children.length
+  const refs = _.map(() => useRef(), _.range(0, count))
```

Comment:
1. Can you just do `_.map(useRef, _.range...)`?
2. Does this cause issues with React if the number of children changes between renders? That would change the number of hooks, right?

Reply:
Hmm, yes I suppose it would. FWIW, I haven't seen any situations in the Terra UI where that is the case. Even if we were to add vertical keyboard navigation to the little 3-item popup menus, the convention seems to be to disable items rather than removing them. So in theory, at least, there will always be the same number.

Is this something we should add support for? I agree it would be more robust if we do, but I'm not sure a good way around it while continuing to use refs.